### PR TITLE
First-cut attempt at a Scala wrapper around proxy-vole

### PR DESCRIPTION
**Pull request by djspiewak**

### CODE REVIEW

**Comment by djspiewak** [comment]
Released as `146.0.0-4f37bd7`

File: foundation/src/main/scala/quasar/contrib/proxy/Search.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.net.ProxySelector
+
+object Search {
+
+  def apply[F[_]: Sync]: F[ProxySelector] =
+    Sync[F] delay {
+      val builder = new ProxySearch
+
+      if (PlatformUtil.getCurrentPlattform() == Platform.WIN) {
```
Comment (djspiewak) [inline foundation/src/main/scala/quasar/contrib/proxy/Search.scala]: I will be forever triggered by `getCurrentPlattform`. `tttttttttttt`

**Comment by jsantos17** [comment]
I tried to implement a version of `quasar-datasource-url`, but it doesn't seem to work. It's always returning `DIRECT` for some reason. [Branch here](https://github.com/jsantos17/quasar-datasource-url/tree/ch9503-proxy-support-through-quasar).

**Comment by djspiewak** [comment]
@jsantos17 That's really really strange. You could try a variant of this PR which *doesn't* override the defaults, perhaps? I noticed that your PR just uses the default proxy-vole search configuration, whereas this PR doesn't.

**Comment by jsantos17** [comment]
Should we debug this further or close this PR?

**Comment by djspiewak** [comment]
> Should we debug this further or close this PR?

Let's debug it further. I want to have a centralized location for this both because of classloader things and because it's going to need to be respected by *every* outbound network connection we make.

**Comment by djspiewak** [comment]
@jsantos17 Merged

**Comment by jsantos17** [comment]
Will merge when green and after testing the URL datasource.

**Comment by jsantos17** [comment]
Travis failed because of [ch9256](https://app.clubhouse.io/data/story/9256/). I really need to get around to fixing that.

**Comment by jsantos17** [comment]
Released as: `147.0.6-1c95bc8`

**Comment by jsantos17** [comment]
Tested this and it's working.

**Pull request merged**